### PR TITLE
Add tree-sitter-zig Go binding into our fork of go-tree-sitter

### PR DESCRIPTION
**Pull request by yangchi**
This changes adds Zig into our fork of go-tree-sitter, thanks to @psarna for pointing me this direction, much better than adding a git_repository of the tree-sitter-zig directly into forge.

The following files are copied from https://github.com/maxxnino/tree-sitter-zig/:
binding.go
binding_test.go
parser.h
parser.c

I made some small changes to a couple of them to make it build + making the style look similar to the rest of tree-sitter-go language subdirs

# Test plan:

go test -v ./...
This passed. (It doesn't test much, just the binding_test.go can work i guess.)

More importantly, I changed the go.mod in my local Forge checkout to point to my local go-tree-sitter and added "initLang(l, zig.GetLanguage(), langs.LangZig)" into queries.go, updated lang.go to MustCreate("zig") as well as Bazel deps. After that bazel build passed in forge. I think that's a good sign (still need a zig.scm to pass test in Forge but that will be a separate PR in forge)

### CODE REVIEW

**Comment by yangchi** [comment]
This doesn't work btw:
`go run ./_automation/main.go update zig`

But I tried a couple other languages, they also do not work, so, i don't know how important this is

**Comment by psarna** [comment]
> This doesn't work btw: `go run ./_automation/main.go update zig`
> 
> But I tried a couple other languages, they also do not work, so, i don't know how important this is

as far as I know, it's just automation for maintainers for updating the parsers. Nothing directly important for runtime though, so we don't care at all

**Review by psarna — APPROVED** [review]

File: _automation/grammars.json [hunk 1]
```diff
@@ -340,5 +340,13 @@
     "reference": "v0.5.0",
     "revision": "6129a83eeec7d6070b1c0567ec7ce3509ead607c",
     "updateBasedOn": "tag"
+  },
+  {
+    "language": "zig",
+    "url": "https://github.com/maxxnino/tree-sitter-zig/",
+    "files":["parser.c"],
+    "reference": "master",
```
Comment (SokolAndrey) [inline _automation/grammars.json]: they use `main` instead. If you change it to main and run update with `--force` it works

Comment (yangchi) [reply _automation/grammars.json]: oh i see. nice. will change

**Review by SokolAndrey — APPROVED** [review]

**Pull request merged**